File: easy/dp/stock.cpp
int maxProfit(vector<int> & prices){
	int n = prices.size();
	if (n == 1)
		return 0;
	int minPrice = prices[0], maxDiff = 0;
	for(auto price : prices){
		minPrice = min(minPrice, price); // minPrice of stock
		maxDiff = max(maxDiff , price - minPrice); // maxDiff of profits of stocks
	}	
	return maxDiff;
}

File: easy/dp/stock.py
def maxProfit(prices):
	n = len(prices)
	if n == 1:
		return 0
	minPrice = prices[0]; maxDiff = 0
	for price in prices:
		minPrice = min(minPrice, price) # minPrice of Stock
		maxDiff = max(maxDiff, price - minPrice) # maxDiff of profits of stocks
	return maxDiff	